lib/services/git-service.ts

```diff
@@ -43,34 +39,31 @@ export class GitService implements IGitService {
         return this.getResourcesFromSource(templateName, assets, version);
     }
 
-    clonePageTemplate(pageName: string, flavor: string, templatesDirectory: string): Promise<string> {
-        const command = "git";
-        const commandArguments: Array<any> = ["clone"];
+    clonePageTemplate(pageName: string, flavor: string, templatesDirectory: string): Promise<any> {
+        return this.getPageTemplatesBaseUrl(flavor)
+            .then((baseUrl: string) => {
+                return new Promise((resolve, reject) => {
+                    const command = "git";
+                    const commandArguments: Array<any> = ["clone"];
 
-        return new Promise((resolve, reject) => {
-            this.getPageTemplatesBaseUrl(flavor)
-                .then((baseUrl: string) => {
                     baseUrl = baseUrl + ".git";
                     commandArguments.push(baseUrl);
                     commandArguments.push(templatesDirectory);
 
                     const process = util.childProcess.spawn(command, commandArguments);
                     process.on("close", (code) => {
                         if (code !== 0) {
-                            return Promise.reject(new Error(`child process exited with code ${code}`));
+                            return reject(new Error(`child process exited with code ${code}`));
                         }
 
                         const pagePath = util.path.join(templatesDirectory, pageName);
                         resolve(pagePath);
                     });
-                })
-                .catch((downloadPageError: any) => {
-                    reject(downloadPageError);
                 });
-        });
+            });
     }
 
-    private getPageTemplatesBaseUrl(flavor: string) {
+    private getPageTemplatesBaseUrl(flavor: string): Promise<any> {
```

Comment:
This method is synchronous, there does not seem any point in making it async to me.

Reply:
In `clonePageTemplate(...)` where it is used we are explicitly creating another Promise anyway so we do not need this async?